src/nvim/lua/treesitter.c

```diff
@@ -1549,22 +1549,105 @@ int tslua_parse_query(lua_State *L)
   return 1;
 }
 
-static const char *query_err_string(TSQueryError err)
+static const char *query_err_to_string(TSQueryError error_type, size_t row, size_t column,
+                                       const char *message)
 {
-  switch (err) {
+  const char *msg;
+  switch (error_type) {
   case TSQueryErrorSyntax:
-    return "invalid syntax";
+    msg = "Invalid syntax:\n";
+    break;
   case TSQueryErrorNodeType:
-    return "invalid node type";
+    msg = "Invalid node type ";
+    break;
   case TSQueryErrorField:
-    return "invalid field";
+    msg = "Invalid field name ";
+    break;
   case TSQueryErrorCapture:
-    return "invalid capture";
+    msg = "Invalid capture name ";
+    break;
   case TSQueryErrorStructure:
-    return "invalid structure";
+    msg = "Impossible pattern:\n";
+    break;
+  default:
+    msg = "error";
+    break;
+  }
+
+  size_t message_length = strlen(msg) + strlen(message) + 40;
```

Comment:
40?

Reply:
Hm 20 proved not to be enough though

Reply:
If you don't know the number then just give it a big one. IOSIZE should do.

Reply:
My comment was more about giving 40 a name. A named constant would help to see what it is doing.

Reply:
For error strings we typically use the same constant as we don't need to optimise for memory along an error path.